ipfs-lite/ipfs_lite.proto

```diff
@@ -0,0 +1,218 @@
+syntax = "proto3";
+
+import "google/protobuf/any.proto";
+
+option java_multiple_files = true;
+option java_package = "io.textile.grpc_ipfs_lite";
+option java_outer_classname = "IpfsLiteProto";
+
+// Model messages
+
+message AddParams {
+    string layout = 1;
+    string chunker = 2;
+    bool rawLeaves = 3;
+    bool hidden = 4;
+    bool shared = 5;
+    bool noCopy = 6;
+    string hashFun = 7;
+}
+
+message Block {
+    bytes rawData = 1;
+    string cid = 2;
```

Comment:
On the threads side, we've been laying into gogo pretty heavily as it has some really nice features around custom types. For example, a `cid.Cid`: https://github.com/textileio/go-textile-threads/blob/master/pb/custom.go#L104. This means that the client doesn't have to manually unmarshal the cid / peerID / threadID, etc. into the final type. 

Proto file usage: https://github.com/textileio/go-textile-threads/blob/master/pb/threads.proto#L75

I'm not sure if that makes sense here though given that the client side probably wont' be able to make use of those types. Just flagging in case it's useful.

Reply:
Interesting. It could make implementing the server nicer maybe? I wasn't aware of this work and don't totally understand it, but we should chat about it soon and see what makes sense.